daemon/watchdog_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package daemon
+
+import (
+	"os"
+	"strconv"
+	"testing"
+)
+
+func TestSdWatchdogEnabled(t *testing.T) {
+	// (time, nil)
+	err := os.Setenv("WATCHDOG_USEC", "100")
+	if err != nil {
+		panic(err)
+	}
+	err = os.Setenv("WATCHDOG_PID", strconv.Itoa(os.Getpid()))
+	if err != nil {
+		panic(err)
+	}
+
+	delay, err := SdWatchdogEnabled()
+	if delay == 0 || err != nil {
+		t.Errorf("TEST: Watchdog enabled FAILED")
```

Comment:
If we're okay with this test being 1.7+ only, we could use [subtests](https://blog.golang.org/subtests) to cleanup these repetitive tests and still have nice error reporting.

I don't know how much this project needs compatibility with older go versions.

Failing at using subtests, I'd still rather have a slice of cases to loop through than this, though It's not a blocking concern.